File: 2019/src/test/java/aoc/Day23Test.java
package aoc;

import org.junit.jupiter.api.*;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day23Test {
    private static String input;
    private Day23 day;

    @BeforeAll
    public static void init() throws IOException {
        input = Files.readString(Paths.get("src/main/resources/day23.txt"), StandardCharsets.UTF_8);
    }

    @BeforeEach
    public void setup() {
        day = new Day23();
    }

    @Nested
    @DisplayName("Part 1")
    class Part1Tests {
        @Test
        void testSolution() {
            long result = day.part1(input);
            assertEquals(23886L, result);
        }
    }

    @Nested
    @DisplayName("Part 2")
    class Part2Tests {
        @Test
        void testSolution() {
            long result = day.part2(input);
            assertEquals(18333L, result);
        }
    }
}

File: 2020/aoc2020/src/test/kotlin/Day23Test.kt
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Nested
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

@DisplayName("Day 23")
class Day23Test {
    private val example = readText("day23_example.txt")
    private val input = readText("day23.txt")
    private val day = Day23()

    @Nested
    @DisplayName("Part 1")
    inner class Part1 {
        @Test
        fun `Matches example 1`() {
            val answer = day.part1(example, 10)
            assertEquals(92_658_374, answer)
        }

        @Test
        fun `Matches example 2`() {
            val answer = day.part1(example, 100)
            assertEquals(67_384_529, answer)
        }

        @Test
        fun `Matches solution`() {
            val answer = day.part1(input, 100)
            assertEquals(47_382_659, answer)
        }
    }

    @Nested
    @DisplayName("Part 2")
    inner class Part2 {
        @Test
        fun `Matches example`() {
            val answer = day.part2(example, 10_000_000)
            assertEquals(149_245_887_792, answer)
        }

        @Test
        fun `Matches solution`() {
            val answer = day.part2(input, 10_000_000)
            assertEquals(42_271_866_720, answer)
        }
    }
}
